File: src/com/github/redfoos/logstash/LogstashLanguage.java
package com.github.redfoos.logstash;

import com.intellij.lang.Language;
import com.intellij.openapi.fileTypes.SingleLazyInstanceSyntaxHighlighterFactory;
import com.intellij.openapi.fileTypes.SyntaxHighlighter;
import com.intellij.openapi.fileTypes.SyntaxHighlighterFactory;
import org.jetbrains.annotations.NotNull;

public class LogstashLanguage extends Language {
    public static final LogstashLanguage INSTANCE = new LogstashLanguage();

    public LogstashLanguage() {
        super("Logstash");
    }
}

File: src/main/kotlin/com/github/redfoos/logstash/LogstashLanguage.kt
package com.github.redfoos.logstash

import com.intellij.lang.Language

class LogstashLanguage private constructor() : Language("Logstash") {

    companion object {
        val INSTANCE = LogstashLanguage()
    }

    override fun isCaseSensitive(): Boolean {
        return true
    }
}
